pandas/core/frame.py

```diff
@@ -2293,14 +2293,19 @@ def _set_item(self, key, value):
         """
 
         self._ensure_valid_index(value)
-        value = self._sanitize_column(key, value)
-        NDFrame._set_item(self, key, value)
-
-        # check if we are modifying a copy
-        # try to set first as we want an invalid
-        # value exeption to occur first
-        if len(self):
-            self._check_setitem_copy()
+        svalue = self._sanitize_column(key, value)
+        try:
+            NDFrame._set_item(self, key, svalue)
+        except com.SettingWithCopyError:
+
+            # if we have a multi-index (which potentially has dropped levels)
+            # need to raise
+            if isinstance(self.is_copy().columns, MultiIndex):
+                raise
+
+            # we have a chained assignment
+            # assign back to the original
+            self.is_copy().loc[self.index,key] = value
```

Comment:
Is there a better name for `is_copy()` if it returns the parent, then `parent()`?